OpenKeychain/src/main/java/org/sufficientlysecure/keychain/receiver/BackupNotificationAlarmReceiver.java

```diff
@@ -0,0 +1,97 @@
+package org.sufficientlysecure.keychain.receiver;
+
+import android.app.AlarmManager;
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+
+import org.sufficientlysecure.keychain.R;
+import org.sufficientlysecure.keychain.provider.KeychainContract;
+import org.sufficientlysecure.keychain.ui.ViewKeyActivity;
+import org.sufficientlysecure.keychain.util.Preferences;
+
+
+public class BackupNotificationAlarmReceiver extends BroadcastReceiver {
+    public static final String MASTER_KEY_ID = "masterKeyId";
+    public static final String NAME = "name";
+    public static final String EMAIL = "email";
+
+    public static final String PERFORM_BACKUP = "performBackup";
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Log.i(BackupNotificationAlarmReceiver.class.getName(),"Inside alarm receiver");
+
+        NotificationManager notificationManager = (NotificationManager)
+                context.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        long masterKeyId = intent.getLongExtra(MASTER_KEY_ID,0);
+        String name = intent.getStringExtra(NAME);
+        String email = intent.getStringExtra(EMAIL);
+
+        // check if backup is already created
+        String masterKeyIds = Preferences.getPreferences(context).getBackupReminders();
```

Comment:
As we are showing a reminder notification after 5 days. But what if the user has already created a backup (let say on 2nd day) then it will show notification unnecessarily after 5 days when the alarm will trigger. To handle this situation I am using preferences.  It will save masterKeyIds in preferences as a key is created (In CreateKeyFinalFragment) and remove them from preferences as the backup is created (in BackupCodeFragment) .  In this way at any point of the time masterKeyIds of all those keys whose backup is not created will be there in preferences. And I will check when the alarm is fired whether masterKeyId of the corresponding key is in the preference or not. If not that means backup is created in the meanwhile so, notification will not be shown.